bin/shell/osh.pl

```diff
@@ -599,6 +652,45 @@ sub main_exit {
     }
 }
 
+my $proxyIp   = undef;
+my $proxyPort = undef;
+my $proxyUser = $user;    # user might be undef. We'll handle that later
+# Parse proxyjump args if specified
+if ($proxyJump) {
+    if ($proxyJump =~ /^(?:([a-zA-Z0-9._@!-]{1,128})@)?(\[?[a-zA-Z0-9._-]+\]?)(?::(\d+))?$/) {
+        $proxyUser = $1 if $1;
+        $proxyIp   = $2;
+        $proxyPort = $3 ? $3 : 22;
+        osh_debug("parsed proxyjump: host=$proxyIp port=$proxyPort user=$proxyUser");
+    }
+    else {
+        main_exit OVH::Bastion::EXIT_INVALID_PROXYJUMP, 'invalid_proxyjump',
+          "Invalid proxyjump specification '$proxyJump', should be [user@]host[:port]";
+    }
+
+    $fnret = OVH::Bastion::get_ip(host => $proxyIp, allowSubnets => 0);
+    if (!$fnret && (($osh_command && $host) || !$osh_command)) {
```

Comment:
```suggestion
    if (!$fnret) {
```
In the context of a proxyjump, there's no notion of `$osh_command` that could be mistaken for a `$host`